### ISSUE TITLE
xdcc get of a previous failed download fails to restart

### ISSUE THREAD

**Issue by DukeLupus** [DukeLupus/dlFilter#71]
Submitter: Joseph
Email: admin@tomobiki.dyndns.org
On the Rizon irc service when trying to restart a failed xdcc batch command the file fails to start. if I disable dlFilter and then start the download it restarts correctly. Downloading new files works just fine only incomplete downloads fail. In options unchecking all the boxes in the FILE REQUESTS section doesn't stop the incomplete download from failing. A work around is to disable dlF, start the incomplete downloads then reenable dlF after the downloads start.

**Comment by Sophist-UK** [comment]
I am trying to work out what is going on but without more detail of the channel and trigger used, whether the channel is set as a DLF filtered channel, exact messages received and maybe even a DLF/Watch log, it is difficult to know exactly what is happening.

It should be noted, however, the DLF does not (currently) recognise XDCC requests (though that would not be too difficult to add), and cannot provide firewall functionality until it does.

**Comment by Sophist-UK** [comment]
Support for `/MSG nick XDCC SEND` added in commit 7bd57313065789bdb3c241d771adb7c082cf8c1b.

**Comment by Sophist-UK** [comment]
I have no explanation as to why DLF is creating the symptoms reported - but clearly it is the cause if disabling it allows the file transfer to work.

To progress this further we need a repeatable test that triggers this issue.

**Comment by Sophist-UK** [comment]
Submitter: Joseph
Email: admin@tomobiki.dyndns.org

I'm running mIRC version 7.52

Here is the rizon host I'm currently connected to welcome message:
```
Welcome to the Rizon Internet Relay Chat Network Lamda
Your host is irc.falconspy.org, running version plexus-4(hybrid-8.1.20)
This server was created Sep 15 2017 at 22:19:46
-
irc.falconspy.org plexus-4(hybrid-8.1.20) CDGNRSUWagilopqrswxyz BCIMNORSabcehiklmnopqstvz Iabehkloqv
-
CALLERID CASEMAPPING=rfc1459 DEAF=D KICKLEN=180 MODES=4 PREFIX=(qaohv)~&@%+ STATUSMSG=~&@%+ EXCEPTS=e INVEX=I NICKLEN=30 NETWORK=Rizon MAXLIST=beI:250 MAXTARGETS=4 are supported by this server
CHANTYPES=# CHANLIMIT=#:250 CHANNELLEN=50 TOPICLEN=390 CHANMODES=beI,k,l,BCMNORScimnpstz WATCH=60 NAMESX UHNAMES AWAYLEN=180 KNOCK ELIST=CMNTU SAFELIST are supported by this server
-
There are 22 users and 13923 invisible on 20 servers
64 IRC Operators online
5 unknown connection(s)
11968 channels formed
I have 13945 clients and 20 servers
-
Current local users: 13945  Max: 18202
Current global users: 13945  Max: 18202
-
- irc.falconspy.org Message of the Day - 
-
-          oo                            
- 
- 88d888b. dP d888888b .d8888b. 88d888b. 
- 88'  `88 88    .d8P' 88'  `88 88'  `88 
- 88       88  .Y8P    88.  .88 88    88 
- dP       dP d888888P `88888P' dP    dP
- 
- Rizon Chat Network -- http://rizon.net
- 
- Listening on ports: 6660 - 6669, 7000. SSL: 6697, 9999
- 
```
Channel is: #news
Command entered is: `/msg Kareraisu xdcc batch 16540,16543,16545`

Normal reply when starting the downloads is : 
```
-> *Kareraisu* xdcc batch 16540,16543,16545
Private: -KareRaisu- ** Sending you pack #16540 ("[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv"), which is 412MB. (resume supported)
Private: -KareRaisu- ** Sending you pack #16543 ("[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv"), which is 309MB. (resume supported)
Reply when trying to restart a failed download with the retry 3 times tab checked: [dlFilter] Running current version of dlFilter
-> *Kareraisu* xdcc batch 16540,16543,16545
Private: -KareRaisu- ** Sending you pack #16540 ("[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv"), which is 412MB. (resume supported)
Private: -KareRaisu- ** Sending you pack #16543 ("[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv"), which is 309MB. (resume supported)
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22878 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22881 26790608
Private: -KareRaisu- ** You can only have 2 transfers at a time, Added you to the main queue for pack 16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv") in position 3. To Remove yourself at a later time type "/MSG KareRaisu XDCC REMOVE 16545".
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22878 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22881 26790608
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** Sending you queued pack #16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv"), which is 313MB. (resume supported)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Reply when trying to restart a failed download with the retry 3 times tab NOT checked: [dlFilter] Running current version of dlFilter
-> *Kareraisu* xdcc batch 16540,16543,16545
Private: -KareRaisu- ** Sending you pack #16540 ("[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv"), which is 412MB. (resume supported)
Private: -KareRaisu- ** Sending you pack #16543 ("[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv"), which is 309MB. (resume supported)
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22881 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22882 26790608
Private: -KareRaisu- ** You can only have 2 transfers at a time, Added you to the main queue for pack 16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv") in position 2. To Remove yourself at a later time type "/MSG KareRaisu XDCC REMOVE 16545".
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22881 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22882 26790608
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** Sending you queued pack #16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv"), which is 313MB. (resume supported)
Private: DCC ACCEPT "[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv" 22881 6907156
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: DCC ACCEPT "[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv" 22881 6907156
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
```
Reply when trying to restart a failed download with all the boxes in the FILE REQUEST of the OTHER section NOT checked:
```
-> *Kareraisu* xdcc batch 16540,16543,16545
Private: -KareRaisu- ** Sending you pack #16540 ("[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv"), which is 412MB. (resume supported)
Private: -KareRaisu- ** Sending you pack #16543 ("[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv"), which is 309MB. (resume supported)
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22877 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22881 26790608
Private: -KareRaisu- ** You can only have 2 transfers at a time, Added you to the main queue for pack 16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv") in position 6. To Remove yourself at a later time type "/MSG KareRaisu XDCC REMOVE 16545".
Private: DCC ACCEPT "[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv" 22877 27960772
Private: DCC ACCEPT "[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv" 22881 26790608
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** Sending you queued pack #16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv"), which is 313MB. (resume supported)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
Private: DCC ACCEPT "[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv" 22877 6907156
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (150 seconds remaining until timeout)
Private: DCC ACCEPT "[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv" 22877 6907156
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (90 seconds remaining until timeout)
Private: -KareRaisu- ** You have a DCC pending, Set your client to receive the transfer. Type "/MSG KareRaisu XDCC CANCEL" to abort the transfer. (30 seconds remaining until timeout)
Private: -KareRaisu- ** Closing Connection: DCC Timeout (180 Sec Timeout)
```
Reply with dlFilter disable:
```
-> *Kareraisu* xdcc batch 16540,16543,16545
-KareRaisu- ** Sending you pack #16540 ("[HorribleSubs] Hyakuren no Haou to Seiyaku no Valkyria - 05 [720p].mkv"), which is 412MB. (resume supported)
-KareRaisu- ** Sending you pack #16543 ("[HorribleSubs] Persona 5 The Animation - 18 [720p].mkv"), which is 309MB. (resume supported)
-KareRaisu- ** You can only have 2 transfers at a time, Added you to the main queue for pack 16545 ("[HorribleSubs] Ongaku Shoujo - 05 [720p].mkv") in position 5. To Remove yourself at a later time type "/MSG KareRaisu XDCC REMOVE 16545".
```
When disabled the downloads resumed at the correct percentage. On the failed retries a second download entry so started so I had a total of 6 download entries for the 3 restarts. The percentage

**Comment by Sophist-UK** [comment]
Thanks for this information which is more helpful. I still need to undertake further research, but here are my current thoughts:

I suspect that this might be an issue of conflict with another script. Do you have any other scripts loaded, especially ones for XDCC support? Please provide me with your list of scripts and URLs where I can download them for review.

Assuming that it is a script incompatibility, using the latest Github version, please try checking the “Other / mIRC-wide / Load dlFilter last (rather than first)” dlFilter option and see if that helps.

**Comment by Sophist-UK** [comment]
This issue has hopefully been fixed in eb0319183b3f2a635fd818b880061a861ee1571e.

I am not planning to support XDCC BATCH unless someone explicitly requests it. If you want to use XDCC BATCH I suggest that you add the server's IRC details to your mIRC trusted user list.

**Comment by Sophist-UK** [comment]
User reports fixes worked.
